### PR TITLE
Enabling focus on people picker remove button and creating support for format strings

### LINKED ISSUE #4816: Focus cannot move to the remove button on people picker selected item
#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? No

Requested priority: Normal

Products/sites affected: ODB Share

#### Describe the issue:

When the <- arrow is hit after multiple people have been selected from the people picker, focus moves to the entire people picker item. The item is not actionable, but the remove button is. Focus should move to the remove button.

### PR DESCRIPTION
**Pull request by rebeccaballantyne**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #4816 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

DO NOT REVIEW yet. I need to work with Ben to talk through the design of multi-selection.

Fixes issue where when using the <- button focus would go to the entire people picker item instead of just the remove button (the only part of the item that should be focusable). Now you can press <- and then "Enter" to remove an item. 

I talked with Ben Truelove about the design for selected items (since previously we would highlight the entire item blue), and the new behavior where we show a focus rect around just the x button is correct.

Also adds support for format strings, so we can have more descriptive aria labels than a generic label - i.e., so "Remove Annie Lindqvest" can be read instead of just "Remove".

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts [hunk 1]
```diff
@@ -114,10 +114,22 @@ export interface IBasePickerProps<T> extends React.Props<any> {
...
+   * is provided, then the primary text of the selected item will be used to fill the string. In this example,
+   * if the selected item is Annie Lindqvest, then the completed aria label would be "Remove Annie Lindqvest".
    * @default ''
    */
   removeButtonAriaLabel?: string;
+  /**
+   * Text that is read after successful removal of selected item. If a format string is provided, then the
+   * primary text of the selected item will be used to fill the string.
+   */
+  singularItemRemovalText?: string;
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts]: Is this mainly for aria, or is it visible?

File: packages/office-ui-fabric-react/src/utilities/string/StringHelper.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Regex that finds {#} so it can be replaced by the arguments in string format
+const FORMAT_REGEX = /\{(\d+)\}/g;
+
+/**
+ * String Format is like C# string format. Usage Example: "hello {0}!".format("world")
+ * will return "hello world!". Calling format on a string with less arguments than
+ * specified in the format will substitute "undefined"
+ */
+export function format(template: string | undefined, ...values: any[]): string {
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/utilities/string/StringHelper.ts]: Don't values need to be strings too?

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx [hunk 3]
```diff
@@ -193,6 +195,7 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
           isInnerZoneKeystroke={ this._isFocusZoneInnerKeystroke }
         >
           <div className={ styles.screenReaderOnly } role='alert' id='selected-suggestion-alert' aria-live='assertive'>{ selectedSuggestionAlert } </div>
+          <div className={ styles.screenReaderOnly } role='alert' id='suggestion-removed-alert' aria-live='assertive'>{ this.state.removedItemAlertText }</div>
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx]: There definitely shouldn't be two live regions. Selected suggestion should win if it's not null, that means the user has changed something and that should be the only region.

**Comment by dzearing** [comment]
This change hasn't been touched in 24 days and contradicts the current design. Let's discuss offline the best direction forward. Closing for now. (It can be reopened once we're clear on the direction.)

**Pull request closed without merging**